pkg/kubelet/eviction/helpers.go

```diff
@@ -1243,6 +1252,23 @@ func evictionMessage(resourceToReclaim v1.ResourceName, pod *v1.Pod, stats stats
 	if !ok {
 		return
 	}
+
+	// Pod level resources will be included in eviction message, along container resources
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) && resourcehelper.IsPodLevelResourcesSet(pod) && resourcehelper.IsSupportedPodLevelResource(resourceToReclaim) {
+		podRequest := v1resource.GetResourceRequestQuantity(pod, resourceToReclaim)
+
+		var podUsage *resource.Quantity
+
+		if resourceToReclaim == v1.ResourceMemory && podStats.Memory != nil {
+			podUsage = memoryUsage(podStats.Memory)
+		}
+
+		message += fmt.Sprintf(podMessageFmt, pod.Name, podUsage.String(), podRequest.String(), resourceToReclaim)
+
+		annotations[OffendingPodKey] = pod.Name
```

Comment:
this seems redundant. The annotations are set on an event that's attached to the pod object.

Reply:
Yes, it is redundant, but added them with the purpose of also focusing on the pod overall, and not only the specific containers. Should I remove it?